sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1229,3 +1229,98 @@ case class Flatten(child: Expression) extends UnaryExpression {
 
   override def prettyName: String = "flatten"
 }
+
+/**
+ * Returns the array containing the given input value (left) count (right) times.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(element, count) - Returns the array containing element count times.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('123', 2);
+       ['123', '123']
+  """)
+case class ArrayRepeat(left: Expression, right: Expression)
+  extends BinaryExpression {
+
+  override def dataType: ArrayType = ArrayType(left.dataType, left.nullable)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val expected = IntegerType
+    if (!expected.acceptsType(right.dataType)) {
+      val mismatch = s"argument 2 requires ${expected.simpleString} type, " +
+        s"however, '${right.sql}' is of ${right.dataType.simpleString} type."
+      TypeCheckResult.TypeCheckFailure(mismatch)
+    } else {
+      TypeCheckResult.TypeCheckSuccess
+    }
+  }
+
+  override def nullable: Boolean = false
```

Comment:
What about cases when `right` is evaluated to `null`?

Reply:
Fixed in 596a54f